plugin/transfer/transfer.go

```diff
@@ -107,11 +108,13 @@ func (t *Transfer) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Ms
 	// Send response to client
 	ch := make(chan *dns.Envelope)
 	tr := new(dns.Transfer)
-	wg := new(sync.WaitGroup)
-	wg.Add(1)
+	closeCh := make(chan struct{})
 	go func() {
-		tr.Out(w, r, ch)
-		wg.Done()
+		err := tr.Out(w, r, ch)
```

Comment:
we could potentially save this error by using a closure and then return it later when closeCh is closed. Also means we don't need to reinvent a new error

Reply:
we should really save this error and return it, no need to make up new error messages